openmdao/docs/features/model_visualization/n2_details.rst

```diff
@@ -0,0 +1,297 @@
+.. _n2_details:
+
+**************************************
+Details of the N2 Model Visualizations
+**************************************
+
+The N2 diagram has a number of sections:
+
+* Model hierarchy
+* Connection matrix
+* Solver structure
+* Toolbar
+* Search bar
+* Legend
+
+This page will go into the details of each section.
+
+Just like in the :ref:`N2 Basics<n2_basics>` section of this documentation, we will use the
+circuit model that is missing a connection as our example code.
+
+.. embed-code::
+    ../test_suite/scripts/circuit_with_unconnected_input.py
+
+Here is the N2 diagram that is generated from this model.
+There are three parts to the N2 diagram. The model hierarchy is on the left, the connection matrix is displayed
+in the center, and the solver structure is shown on the right.
+
+
+.. raw:: html
+    :file: examples/n2_circuit_with_unconnected_input.html
+
+
+Model Hierarchy
+---------------
+
+The model hierarchy is on the left side of the diagram. It is actually a mix of Systems and variables, both
+inputs and outputs.
+
+The top System, the root of the model, is on the left and the subsystem hierarchy is displayed to right.
+
+The colors indicate the type of
+item in the hierarchy. To see what the colors represent, an optional legend can be displayed using the |show_legend| button.
+Here is the part of the legend that explains the colors of the model hierarchy.
+
+.. image::
+    images/systems_and_variables_legend.png
+
+You can click in the model hierarchy to collapse and uncollapse sections of the hierarchy. Left clicking on a
+subsystem zooms the hierarchy display to just that subsystem in the hierarchy. Right clicking on a subsystem collapses all of the
+hierarchy to the right of it.
+
+Connection Matrix
+-----------------
+The matrix in the middle of the diagram shows what items in the model are connected.
+
+The color of an item on a diagonal cell matches the color of the rightmost element in model hierarchy. The colors
+for the off-diagonal items, which indicate connections, are always black.
+
+The symbol of an item in the connection matrix indicates whether the variable is a scalar, vector, or collapsed variable.
+
+.. image::
+    images/connection_matrix_legend.png
+
+Black boxes outline items in the matrix to show the grouping of the variables of the lowest level components.
+
+When you hover over a connection in the matrix, arrows are drawn to indicate connections. The behavior is different
+depending on whether you hover over a diagonal or an off-diagonal cell.
+
+When hovering over a diagonal item, the explicit connections to/from that variable are displayed. Red arrows indicate
+that element is the target of the connection. A green arrow indicates it is the source.
+
+When hovering over a off-diagonal, the behavior differs depending on whether the cell is in the lower or upper
+triangular part of the diagram.
+
```

Comment:
"When hovering over a off diagonal" --> "When hovering over an off-diagonal connection,".

Also, I think we should probalby make clear that this is hovering over a black square or dot.  You might also want to describe the behavior when you hover over a non-existent connection (where the line is blue).

Reply:
FIxed the "a" -> "an".

I didn't want to document the blue lines because that's part of the draw_potential_connections feature that we want to get rid of. There is a story for that. I should probably do that one right away. It's easy.